src/emmetHelper.ts

```diff
@@ -949,37 +949,56 @@ function getFormatters(syntax: string, preferences: any) {
 
 /**
  * Updates customizations from snippets.json and syntaxProfiles.json files in the directory configured in emmet.extensionsPath setting
+ * @param emmetExtensionsPathSetting setting passed from emmet.extensionsPath, support multiple path setting
+ * https://github.com/microsoft/vscode/issues/116741
  */
-export async function updateExtensionsPath(emmetExtensionsPath: string | undefined | null, fs: FileService, workspaceFolderPath?: URI, homeDir?: URI): Promise<void> {
-	if (emmetExtensionsPath) {
-		emmetExtensionsPath = emmetExtensionsPath.trim();
-	}
-	if (!emmetExtensionsPath) {
-		resetSettingsFromFile();
-		return Promise.resolve();
+export async function updateExtensionsPath(emmetExtensionsPathSetting: string | string[] | undefined | null, fs: FileService, workspaceFolderPath?: URI, homeDir?: URI): Promise<void> {
+	let emmetExtensionsArray: string[];
+
+	if (Array.isArray(emmetExtensionsPathSetting)){
+		emmetExtensionsArray = emmetExtensionsPathSetting;
+	} else {
+		emmetExtensionsArray = [emmetExtensionsPathSetting];
 	}
 
 	let emmetExtensionsPathUri: URI | undefined;
-	if (emmetExtensionsPath[0] === '~') {
-		if (homeDir) {
-			emmetExtensionsPathUri = joinPath(homeDir, emmetExtensionsPath.substr(1));
+	let findValidPath = false;
+	for (let emmetExtensionsPath of emmetExtensionsArray){
+		if (emmetExtensionsPath) {
+			emmetExtensionsPath = emmetExtensionsPath.trim();
 		}
-	} else if (!isAbsolutePath(emmetExtensionsPath)) {
-		if (workspaceFolderPath) {
-			emmetExtensionsPathUri = joinPath(workspaceFolderPath, emmetExtensionsPath);
+		if (!emmetExtensionsPath) {
+			resetSettingsFromFile();
+			return Promise.resolve();
 		}
-	} else {
-		emmetExtensionsPathUri = URI.file(emmetExtensionsPath);
-	}
 
-	try {
-		// the fs.stat call itself could throw, so we wrap this part up into a try-catch
-		if (!emmetExtensionsPathUri || (await fs.stat(emmetExtensionsPathUri)).type !== FileType.Directory) {
-			throw new Error();
+		if (emmetExtensionsPath[0] === '~') {
+			if (homeDir) {
+				emmetExtensionsPathUri = joinPath(homeDir, emmetExtensionsPath.substr(1));
+			}
+		} else if (!isAbsolutePath(emmetExtensionsPath)) {
+			if (workspaceFolderPath) {
+				emmetExtensionsPathUri = joinPath(workspaceFolderPath, emmetExtensionsPath);
+			}
+		} else {
+			emmetExtensionsPathUri = URI.file(emmetExtensionsPath);
 		}
-	} catch (e) {
+
+		try {
+			// the fs.stat call itself could throw, so we wrap this part up into a try-catch
+			if (!emmetExtensionsPathUri || (await fs.stat(emmetExtensionsPathUri)).type !== FileType.Directory) {
+				throw new Error();
+			}
+		} catch (e) {
+			continue;
+		}
+		findValidPath = true;
+		break;
+	}
+
+	if (!findValidPath){
 		resetSettingsFromFile();
-		throw new Error(`The directory ${emmetExtensionsPath} doesn't exist. Update emmet.extensionsPath setting`);
+		throw new Error(`The directory ${emmetExtensionsPathSetting} doesn't exist. Update emmet.extensionsPath setting`);
```

Comment:
Just noticed this string should be localized (see https://github.com/microsoft/vscode-emmet-helper/blob/main/src/emmetHelper.ts#L131), and now we have both singular and plural cases since users can pass in an array, so determining how to restate this error is another challenge.

Reply:
I'm looking into it! Should we restate the error in this PR or should we create another one?

Reply:
The end goal is just to let the user know something is wrong with the way they set extensionPath. Whether or not it deserves multiple errors, or different forms of errors, is up to you. I guess my personal approach would be to use another `Array.isArray` check, and write two similarly-worded exceptions, to resolve the single/plural issue. There might be other ways, though.